pydantic/_internal/_decorators.py

```diff
@@ -218,3 +256,127 @@ def in_ipython() -> bool:
         return False
     else:  # pragma: no cover
         return True
+
+
+class OnlyValueValidator(Protocol):
+    def __call__(self, __value: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValues(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesKwOnly(Protocol):
+    def __call__(self, __value: Any, *, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargs(Protocol):
+    def __call__(self, __value: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargsAndValue(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesAndKwargs(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+V1Validator = Union[
+    V1ValidatorWithValues, V1ValidatorWithValuesKwOnly, V1ValidatorWithKwargs, V1ValidatorWithValuesAndKwargs
+]
+
+
+def make_generic_validator(
+    validator: V1Validator | OnlyValueValidator | ValidatorFunction, mode: str
+) -> ValidatorFunction:
+    """
+    Make a generic function which calls a validator with the right arguments.
+    """
+    sig = signature(validator)
+
+    def _warn_v1_validator() -> None:
+        warnings.warn(
+            'Validator signatures using the `values` keyword argument or `**kwargs` are no longer supported.'
+            ' Please use an `info: pydantic.ValidationInfo` as the second positional argument instead.'
+            ' This compatibility shim may be removed in a future minor release of Pydantic v2.X',
+            DeprecationWarning,
+            # The stacklevel parameter makes the warning show up as coming from the User's code instead
+            # of our internal implementation
+            # Since this just goes up the stack the source location will appear as
+            # class TheModelName(BaseModel):
+            # Which is good enough for now in terms of helping users locate the issue
+            # In the future maybe we can capture the calling module and line number in the
+            # @validator decorator and use that here to be more accurate about where the issue is
+            # But that may not be 100% reliable, so tabling for now.
+            stacklevel=6,
+        )
+
+    positional_params: List[str] = []
+    keyword_only_params: List[str] = []
+    accepts_kwargs = False
+    for param_name, parameter in sig.parameters.items():
+        if param_name in ('field', 'config'):
+            raise TypeError(
+                'The `field` and `config` parameters are not available in Pydantic V2.'
+                ' Please use the `info` parameter instead.'
+                ' You can access the configuration via `info.config`,'
+                ' but it is a dictionary instead of an object like it was in Pydantic V1.'
+                ' The `field` argument is unavailable.'
```

Comment:
```suggestion
                ' The `field` argument is no longer available.'
```
I think this makes it more clear that it was purposely removed. I think we will need to document how to migrate if people were using this in their own v1 validators — is there a migration path for this? If not, do we need to add this back in?

Reply:
I just read https://github.com/pydantic/pydantic/issues/5170 — I think if we adopt the approach Samuel described there, it's probably worth updating this message to indicate how to access the relevant data (i.e., using `cls.model_fields`).